src/advanced_inv.cpp

```diff
@@ -1619,6 +1619,7 @@ void advanced_inventory::display()
                     int moved = 0;
                     for( auto &elem : moving_items ) {
                         assert( !elem.is_null() );
+                        move_cost += ( elem.count_by_charges() ? 1 : elem.charges ) * Pickup::cost_to_move_item( g->u, elem );
```

Comment:
I think it's because of this line. It should not multiply anything since `elem` is a single item. Changing it to `move_cost += Pickup::cost_to_move_item( g->u, elem )` should fix it.

Reply:
Additionally, items count by charge should have their movement cost scale with `item charge / default charge size`, maybe in `Pickup::cost_to_move_item`.